dateutil/tz/tz.py

```diff
@@ -1596,7 +1596,7 @@ def nocache(name=None):
                     name = os.environ["TZ"]
                 except KeyError:
                     pass
-            if name is None or name == ":":
```

Comment:
I'll note that this will suddenly allow `tz.gettz(0)` without throwing an exception.

Maybe we should go with `if name is None or name in ("", ":"):`?

In a potential future version we may want to explicitly raise `TypeError` if `name` is not an instance of `str` or `None`, but I'm inclined to defer the discussion about duck typing/leniency and just implement this in such a way that we don't have to address it right now.

Reply:
Sure, makes sense. Although I would be interested to hear what you think about how we should do typing (#383) after the deprecation #653 

Also, should I squash my commits to clean up the commit history, will you just squash them all when merging, or what are your preferences in that regard?

Reply:
I personally rewrite my history so that each commit is an atomic change where the tests pass. If that's too much of a pain, squashing it down is also a valid approach.